src/dnaio/_core.pyx

```diff
@@ -337,76 +337,81 @@ cdef class FastqIter:
         self.extra_newline = False
         self.yielded_two_headers = False
         self.eof = False
-        self.bufend = 0
         self.record_start = 0
         self.file = file
         if buffer_size < 1:
             raise ValueError("Starting buffer size too small")
 
+    def __dealloc__(self):
+        if self.buffer != NULL:
```

Comment:
Strictly speaking, you don’t need to explicitly check for a null pointer because `PyMem_Free` will just not do anything if it gets passed one.

Reply:
Fixed. Thanks!